service-base/src/main/java/org/eclipse/hono/service/metric/Metrics.java

```diff
@@ -106,11 +131,57 @@ void reportTelemetry(
             MetricsTags.TtdStatus ttdStatus,
             Sample timer);
 
+    /**
+     * Reports a telemetry message or event received from a device.
+     *
+     * @param type The type of message received, e.g. <em>telemetry</em> or <em>event</em>.
+     * @param tenantId The tenant that the device belongs to.
+     * @param tenantObject The tenant configuration object.
+     * @param outcome The outcome of processing the message.
+     * @param qos The delivery semantics used for sending the message downstream.
+     * @param payloadSize The number of bytes contained in the message's payload.
+     * @param ttdStatus The outcome of processing the TTD value contained in the message.
+     * @param timer The timer indicating the amount of time used
+     *              for processing the message.
+     * @throws NullPointerException if any of the parameters are {@code null}.
+     * @throws IllegalArgumentException if type is neither telemetry nor event or
+     *                    if payload size is negative.
+     */
+    void reportTelemetry(
+            MetricsTags.EndpointType type,
+            String tenantId,
```

Comment:
obsolete

Reply:
yes, it needs to be there until all the adapters use the new method. Shall I mark them as deprecated or remove them ?

Reply:
I think that all adapters need to be changed to use the new variant of this method ...

Reply:
How about the old variant of this method? Should we mark them as deprecated or remove them after all adapters have been changed to use the new variant?

Reply:
I think we should remove it

Reply:
@sophokles73 The obsolete variant of the method `reportCommand(...) `  in `Metrics` interface has been removed as part of the latest commits pushed. The method `reportTelemetry(..) `is still being in use in COAP Adpater. It will be removed in the PR for the issue #1330 for COAP Adapter.